e2e_tests/tests/cluster/test_agent_disable.py

```diff
@@ -83,10 +85,13 @@ def _disable_agent(agent_id: str, drain: bool = False, json: bool = False) -> It
         + [agent_id]
     )
     try:
-        yield subprocess.check_output(command).decode()
+        with logged_in_user(ADMIN_CREDENTIALS):
+            out = subprocess.check_output(command).decode()
+        yield out
```

Comment:
I thought so too (my previous commit is me trying that)

```py
from contextlib import contextmanager

@contextmanager
def a():
    print("Open a")
    yield
    print("Closing a")

@contextmanager
def test():
    with a():
        yield

with test():
    print("Inside test")
 ```
 
But this surprising outputs
 ```
 Open A
 Inside test
 Closing a
 ```

Reply:
maybe I am missing something, but what's wrong with that? 

```
from contextlib import contextmanager

@contextmanager
def a():
    print("Open a")
    yield "result"
    print("Closing a")

@contextmanager
def test():
    print("open test")
    with a() as a_res:
        yield a_res + " appended"
    print("close test")

with test() as test_result:
    print("Inside test", test_result)
```
prints
```
open test
Open a
Inside test result appended
Closing a
close test
```

Reply:
My bad I wasn't being very clear at how this relates to the disable agent.

It was causing some test failures due to everything in the ```with _disable_agent(...)``` context being ran as admin since the logout doesn't execute until the ```_disable_agent(...)``` context is finished.

So this test starts  ```command_id1``` outside of a ```_disable_agent(...)``` context then reads the status of the command inside (which it fails since the command was ran as ```determined```).

https://github.com/determined-ai/determined/blob/25b29f379ee79825d2edbc6348237a7635269a28/e2e_tests/tests/cluster/test_agent_disable.py#L290-L303

Reply:
oh makes total sense. disregard my comment then.